frontend/src/components/Patchnotes.jsx

```diff
@@ -56,9 +65,96 @@ const Patchnotes = ({ game }) => {
             }
         };
 
-        fetchPatchDetails();
+        fetchPatchnotes();
     }, [game, year, month, version]);
 
+    const submitComment = async (comment) => {
+        try {
+            let url = '';
+            if (game === 'overwatch') {
+                url = `http://localhost:3000/patchnotes/overwatch/${year}/${month}/comments`;
+            } else {
+                url = `http://localhost:3000/patchnotes/valorant/${version}/comments`;
+            }
+
+            const response = await fetch(url, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify(comment),
+            });
+            if (!response.ok) {
+                throw new Error('Failed to create comment');
+            }
+            const newComment = await response.json();
+            newComment.user = { username: user.username };
+            setPatchnotes((prevPatchnotes) => ({
+                ...prevPatchnotes,
+                comments: [...prevPatchnotes.comments, newComment],
+            }));
+        } catch (error) {
+            console.error(error);
+        }
+    };
+
+    const deleteComment = async (commentId) => {
+        try {
+            let url = '';
+            if (game === 'overwatch') {
+                url = `http://localhost:3000/patchnotes/overwatch/${year}/${month}/${commentId}`;
+            } else {
+                url = `http://localhost:3000/patchnotes/valorant/${version}/${commentId}`;
+            }
+
+            const response = await fetch(url, {
+                method: 'DELETE',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+            });
+            if (!response.ok) {
+                throw new Error('Failed to delete comment');
+            }
+            setPatchnotes((prevPatchnotes) => ({
```

Comment:
what are we setting here? what is the purpose of this `setPatchnotes` call?

Reply:
It redefines the patchnotes array, and deletes the selected comment by filtering based on id.